objects/src/accounts/seed.rs

```diff
@@ -0,0 +1,65 @@
+use super::{parse_felt, AccountError, AccountId, Hasher, Word};
+
+// ACCOUNT SEED GENERATOR
+// ================================================================================================
+
+/// Specifies the account type.
+pub(super) enum AccountType {
+    FungibleFaucet,
+    NonFungibleFaucet,
+    RegularAccountImmutableCode,
+    RegularAccountUpdatableCode,
+}
+
+/// Finds and returns a seed suitable for creating account IDs for the specified account type
+pub(super) fn get_seed(
+    init_seed: [u8; 32],
+    account_type: AccountType,
+    on_chain: bool,
+) -> Result<Word, AccountError> {
+    let mut current_seed: Word = [
+        parse_felt(&init_seed[..8])?,
+        parse_felt(&init_seed[8..16])?,
+        parse_felt(&init_seed[16..24])?,
+        parse_felt(&init_seed[24..32])?,
+    ];
+    let mut current_digest = Hasher::hash_elements(&current_seed);
+
+    // loop until we have a seed that satisfies the specified account type.
+    loop {
+        // check if the seed satisfies the specified account type
+        if let Ok(_) = AccountId::validate_seed_digest(&current_digest) {
+            if let Ok(account) = AccountId::try_from(current_digest[0]) {
+                match account_type {
+                    AccountType::FungibleFaucet => {
+                        if account.is_fungible_faucet() && account.is_on_chain() == on_chain {
+                            return Ok(current_seed);
+                        }
+                    }
+                    AccountType::NonFungibleFaucet => {
+                        if account.is_non_fungible_faucet() && account.is_on_chain() == on_chain {
+                            return Ok(current_seed);
+                        }
+                    }
+                    AccountType::RegularAccountUpdatableCode => {
+                        if account.is_regular_account_updatable_code()
+                            && account.is_on_chain() == on_chain
+                        {
+                            println!("{account}");
+                            return Ok(current_seed);
+                        }
+                    }
+                    AccountType::RegularAccountImmutableCode => {
+                        if account.is_regular_account_immutable_code()
+                            && account.is_on_chain() == on_chain
+                        {
+                            return Ok(current_seed);
+                        }
+                    }
+                }
+            }
+        }
+        current_seed = current_digest.into();
+        current_digest = Hasher::hash_elements(&current_seed);
```

Comment:
Instead of re-hashing, we can probably just do `current_seed[0] += ONE;` Should make finding a seed much faster.

Reply:
So after some thinking I think this pattern is more efficient.  We have to calculate the hash digest associated with a seed so we can check if the digest is valid and we have the required POW.  Seeing as we have calculate the digest anyway we can just use this for free as a new "random" seed if it does not satisfy the requirements.

Reply:
Ah - makes sense! I should have read the code more carefully :)